### ISSUE TITLE
Sidebar links not activating properly

### ISSUE THREAD

**Issue by oti215** [creativetimofficial/ct-vue-paper-dashboard-pro#59]
Hello everyone!

After freshly unzipping the project, `npm install` at root failed, because it complains about the package.json not being valid, although I see nothing wrong with its syntax. 

I did, however, pass its dependencies to the package.json in the project im working on (laravel) and was able to install all of them normally. Not sure why yours didn't work.

However, the more inconvenient issue is that im receiving an error in /components/UIComponents/SidebarPlugin/Sidebar.vue :
![image](https://user-images.githubusercontent.com/10524374/43695193-1cc720b4-9905-11e8-823d-12eeff398334.png)

Complaining about an unexpected token, not sure whats going on since the code is fine.

Any ideas what may be causing it ?

**Comment by cristijora** [comment]
Hello @oti215 

Thanks for using our products and for reporting here.
First I'd like to ask which version are you using? (1 or 2)
A common issue along with `zip` archives is that people on MAC OS don't have `show hidden files` enabled by default and therefore when unzipping, don't copy the `.babelrc` and `.eslintrc` files which are quite important. Please make sure you have these files in your project. 

If that is not the problem, then most likely it's because of a different setup (laravel). Make sure your setup supports dynamic imports `import()` syntax. Depending on laravel-mix setup or what you have there the configuration might vary.

**Comment by oti215** [comment]
Hello. I am using version 2. And yes, I'm on a mac. So I set hidden files to show to show and then it allowed me to `npm install`, thanks! 

However, Im not sure how I'm supposed to integrate the theme's dependencies into my own project. 

My vue project has subdomains, 2 for now, but it will have 5 in total. So the structure is a little different for the set up of each SPA since they are completely different apps (clients/admin). I ended copying most dependencies into my package.json in my laravel project, except for `@vue/cli-service` since laravel comes with core-env for building the proyect,  I also passed your `.babelrc` and `eslintrc` into my root folder. I created a structure which allows me to share UIComponents across both apps and keep their custom components separated and easy to access. 

When attempting to build using your `.babelrc` (which I put into my laravel project) I rand into some errors with the "components" plugin declared in there, I ended up removing the content of `.babelrc` and added my own [preset](https://www.npmjs.com/package/babel-preset-vue), and a single [plugin](https://babeljs.io/docs/en/babel-plugin-syntax-dynamic-import/) which solved the `import()` error.  

So I _did_ solve the initial issue which was the dynamic import not working. 

But I ran into another problem (let me know if i should create a new issue with this). Whenever I click on a SideBar menu item with a drop down, I get the following error:
![image](https://user-images.githubusercontent.com/10524374/43966712-9e9b42d0-9c90-11e8-899b-0ed44deea8da.png)

As Im not sure if its an issue with my dependencies, so here's a picture of my package.json:
![image](https://user-images.githubusercontent.com/10524374/43966751-b6313a76-9c90-11e8-8abb-615f46790f04.png)

And my .babelrc:
![image](https://user-images.githubusercontent.com/10524374/43966788-d3fb33a4-9c90-11e8-8c87-6f17cddce09d.png)

Maybe that has something to do with the plugins I removed from `.babelrc` ? 

Excuse me if something is not clear! Just ask and i'll be glad to clear it up.

**Comment by cristijora** [comment]
The problem is that `@vue/cli-service` comes up with some stuff integrated including babel plugins and all that stuff. If you're starting from scratch now, I'd recommend this for laravel https://github.com/yyx990803/laravel-vue-cli-3 
Or you could take a look at this file, https://github.com/yyx990803/laravel-vue-cli-3/blob/master/frontend/vue.config.js and simply copy paste the whole dashboard inside the a `frontend` folder for example and add the additional changes from here https://github.com/yyx990803/laravel-vue-cli-3/blob/master/frontend/vue.config.js to `vue.config.js` 

Otherwise, you'll have to maintain laravel mix configs and babel plugins. 
Regarding the sidebar error, I'm almost sure it's not really related to the dashboard because the Sidebar has not jquery dependency whatsoever. The only place where jquery is used, is the full calendar and the vector maps which are isolated in `vue` components. Must be something else there that is causing the problem.

Let me know if that helps.

**Comment by oti215** [comment]
Took me a day, but I managed to set up my multi page app with vue-cli 3 for building and deploying my apps and assets. Its actually much better than the standard Laravel way of doing it, thanks for the tip. 

After doing this I was still seeing the same error, but this time it was from the /node_modules/jquery package and since I'm not using it I just removed it and it solved that issue.

Im guessing that if I want jquery later on ill have to use it the old fashioned way (cdn or include the .js file) instead of via npm.

**Comment by cristijora** [comment]
Hey @oti215 can you share the code for declaring those sidebar items and also the routes from `routes.js` for these 2 routes. From, what I see the problem is that both routes are considered as active because they are `children`. You can try changing
`linkActiveClass: 'active'` to `linkExactActiveClass: 'active'` in `main.js` and see if that works for you

**Comment by oti215** [comment]
Hello, I tried `linkExactActiveClass: 'active'` with everything as is, but it had no effect. 

The two items are definitely children of the same dropdown. Here are the files

sidebarLinks.js:
![image](https://user-images.githubusercontent.com/10524374/44293901-cc6aa700-a25d-11e8-8768-db0aa8afeb94.png)

routes.js:
![image](https://user-images.githubusercontent.com/10524374/44293944-4ef36680-a25e-11e8-8c0c-db5d8d74594e.png)

In my case, I split the your main.js into two files so I can reuse it in other apps. So theres my app.js which contains app specific declarations, and includes the global stuff from boostrap.js:
![image](https://user-images.githubusercontent.com/10524374/44293969-911ca800-a25e-11e8-9c8a-419dc87ecc3b.png)


And bootstrap.js:
Hello, I tried `linkExactActiveClass: 'active'` with everything as is, but it had no effect. 

The two items are definitely children of the same dropdown. Here are the files

sidebarLinks.js:
![image](https://user-images.githubusercontent.com/10524374/44293901-cc6aa700-a25d-11e8-8768-db0aa8afeb94.png)

routes.js:
![image](https://user-images.githubusercontent.com/10524374/44293944-4ef36680-a25e-11e8-8c0c-db5d8d74594e.png)

In my case, I split the your main.js into two files so I can reuse part of it in other apps. So I have an app.js which contains app specific declarations, boostrap.js which includes the global config stuff.

This is my app.js:
![image](https://user-images.githubusercontent.com/10524374/44293969-911ca800-a25e-11e8-9c8a-419dc87ecc3b.png)

And bootstrap.js:
![image](https://user-images.githubusercontent.com/10524374/44299477-93195200-a2c4-11e8-95cf-ab81f64690ee.png)
Theres more configuration stuff at the bottom of bootstrap.js where I set up axios and a common header which contains the csrf token etc... Im only screenshotted the relevant part of the code. 

Here is my folder structure just in case it helps:
![image](https://user-images.githubusercontent.com/10524374/44299511-906b2c80-a2c5-11e8-979a-f6c92f9cb738.png)

**Comment by oti215** [comment]
Hello, I tried setting `linkExactActiveClass: 'active'` to no avail.

The menu items are in fact children of the same dropdown item. 

Here are the corresponding files:

sidebarLinks.js:
![image](https://user-images.githubusercontent.com/10524374/44553172-a9774180-a6fa-11e8-8722-64a3323e2876.png)

routes.js:
![image](https://user-images.githubusercontent.com/10524374/44553200-b98f2100-a6fa-11e8-86ba-0a12a8cd333f.png)

And app.js, where I declare most of the app specific stuff (the general app stuff like css files are required in from bootstrap.js):
![image](https://user-images.githubusercontent.com/10524374/44553257-e6433880-a6fa-11e8-92bd-6b08600147c6.png)

**Comment by oti215** [comment]
Let me know if you rather have code instead of pictures

**Comment by oti215** [comment]
Still waiting on a response on this...

**Comment by oti215** [comment]
@cristijora @timcreative please assist ?

**Comment by oti215** [comment]
Problem was in sidebarlinks.js the path must be the full path you want to active :T 🤦‍♂️
